android-core/src/main/java/com/mparticle/identity/IdentityApi.java

```diff
@@ -348,6 +350,26 @@ private void reset() {
         }
     }
 
+    private  String generateHash(String input) {
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-256");
+            byte[] hashBytes = digest.digest(input.getBytes());
+
+            // Convert byte array to hex string
+            StringBuilder hexString = new StringBuilder();
+            for (byte b : hashBytes) {
+                String hex = Integer.toHexString(0xff & b);
+                if (hex.length() == 1) hexString.append('0');
+                hexString.append(hex);
+            }
+            return hexString.toString();
+        } catch (NoSuchAlgorithmException e) {
+            //throw new RuntimeException("Hashing algorithm not found", e);
```

Comment:
This feels like an anti-pattern... What do we do if the device doesn't have SHA-256 available? Can this be rewritten to use fallback hashes like SHA-1 or others?